src/Compilers/CSharp/Test/Syntax/Parsing/RoleParsingTests.cs

```diff
@@ -1313,7 +2441,721 @@ public void RoleParsing_WithoutBody_WithoutBaseList_WithTypeParameter(bool isExt
+                    N(SyntaxKind.VariableDeclaration);
+                    {
+                        N(SyntaxKind.PredefinedType);
+                        {
+                            N(SyntaxKind.IntKeyword);
+                        }
+                        N(SyntaxKind.VariableDeclarator);
+                        {
+                            N(SyntaxKind.IdentifierToken, "Y");
+                            N(SyntaxKind.EqualsValueClause);
+                            {
+                                N(SyntaxKind.EqualsToken);
+                                N(SyntaxKind.NumericLiteralExpression);
+                                {
+                                    N(SyntaxKind.NumericLiteralToken, "0");
+                                }
+                            }
+                        }
+                    }
+                    N(SyntaxKind.SemicolonToken);
+                }
+                N(SyntaxKind.CloseBraceToken);
+            }
+            N(SyntaxKind.EndOfFileToken);
+        }
+        EOF();
+    }
+
+    [Fact]
+    public void WithMember_Const_WithoutImplicitOrExplicit()
+    {
+        var text = $$"""extension X for int { const int Y = 0; }""";
+        UsingTreeWithCSharpNext(text);
+
+        N(SyntaxKind.CompilationUnit);
+        {
+            N(SyntaxKind.ExtensionDeclaration);
+            {
+                N(SyntaxKind.ExtensionKeyword);
+                N(SyntaxKind.IdentifierToken, "X");
+                N(SyntaxKind.ForType);
+                {
+                    N(SyntaxKind.ForKeyword);
+                    N(SyntaxKind.PredefinedType);
+                    {
+                        N(SyntaxKind.IntKeyword);
+                    }
+                }
+                N(SyntaxKind.OpenBraceToken);
+                N(SyntaxKind.FieldDeclaration);
+                {
+                    N(SyntaxKind.ConstKeyword);
+                    N(SyntaxKind.VariableDeclaration);
+                    {
+                        N(SyntaxKind.PredefinedType);
+                        {
+                            N(SyntaxKind.IntKeyword);
+                        }
+                        N(SyntaxKind.VariableDeclarator);
+                        {
+                            N(SyntaxKind.IdentifierToken, "Y");
+                            N(SyntaxKind.EqualsValueClause);
+                            {
+                                N(SyntaxKind.EqualsToken);
+                                N(SyntaxKind.NumericLiteralExpression);
+                                {
+                                    N(SyntaxKind.NumericLiteralToken, "0");
+                                }
+                            }
+                        }
+                    }
+                    N(SyntaxKind.SemicolonToken);
+                }
+                N(SyntaxKind.CloseBraceToken);
+            }
+            N(SyntaxKind.EndOfFileToken);
+        }
+        EOF();
+    }
```

Comment:
I'd like to see some tests covering:
* Extensions named `extension`
* Explicitly-implemented implicit and explicit operators implemented:
    * From an interface named `extension`.
    * From an interface in a namespace named `extension`.
* Underlying types of contextual keywords, such as `dynamic` and `var.
* `for` after the base list.